ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -632,10 +670,11 @@ getCompletions plugins ideOpts CC {allModNamesAsNS, anyQualCompls, unqualCompls,
               ty = showForSnippet <$> typ
               thisModName = Local $ nameSrcSpan name
 
-          compls = if T.null prefixModule
-            then map (notQual,) localCompls ++ map (qual,) unqualCompls ++ ((notQual,) . ($Nothing) <$> anyQualCompls)
-            else ((qual,) <$> Map.findWithDefault [] prefixModule (getQualCompls qualCompls))
-                 ++ ((notQual,) . ($ Just prefixModule) <$> anyQualCompls)
+          compls
+            | T.null prefixScope = map (notQual,) localCompls ++ map (qual,) unqualCompls ++ ((notQual,) . ($ Nothing) <$> anyQualCompls)
+            | not $ null recordDotSyntaxCompls = recordDotSyntaxCompls
+            | otherwise = ((qual,) <$> Map.findWithDefault [] prefixScope (getQualCompls qualCompls))
+                 ++ ((notQual,) . ($ Just prefixScope) <$> anyQualCompls)
```

Comment:
Should we union these two sources of completions when a prefix is present?

Reply:
Worth an explanation regardless!

Reply:
The thinking here was that if there are record completions, they will be the only valid completions (since we get record completions from `myrecord.var`, and everything else would only work with something like `myrecord .somefn`, with a space, in which case record completions is null). I'll add a comment. Does that make sense though?